impl/src/main/java/io/kroxylicious/testing/kafka/invm/InVMKafkaCluster.java

```diff
@@ -113,17 +115,31 @@ private Server buildKafkaServer(KafkaClusterConfig.ConfigHolder c) {
         if (kraftMode) {
             var clusterId = c.getKafkaKraftClusterId();
             var directories = StorageTool.configToLogDirectories(config);
-            var metaProperties = StorageTool.buildMetadataProperties(clusterId, config);
-            // note ignoreFormatter=true so tolerate a log directory which is already formatted. this is
-            // required to support start/stop.
-            StorageTool.formatCommand(LOGGING_PRINT_STREAM, directories, metaProperties, MINIMUM_BOOTSTRAP_VERSION, true);
+            prepareLogDirsForKraft(clusterId, config, directories);
             return instantiateKraftServer(config, threadNamePrefix);
         }
         else {
             return new KafkaServer(config, Time.SYSTEM, threadNamePrefix, false);
         }
     }
 
+    private static void prepareLogDirsForKraft(String clusterId, KafkaConfig config, Seq<String> directories) {
+        try {
+            // in kafka 3.7.0 the MetadataProperties class moved package, we use reflection to enables the extension to work with
+            // the old and new class.
+            Method buildMetadataProperties = StorageTool.class.getMethod("buildMetadataProperties", String.class, KafkaConfig.class);
+            Object metaProperties = buildMetadataProperties.invoke(null, clusterId, config);
+            Method formatCommand = Arrays.stream(StorageTool.class.getMethods()).filter(method -> method.getName().equals("formatCommand")
+                    && method.getParameterCount() == 5).findFirst().orElseThrow();
```

Comment:
bit sad, we can't look up the method directly like we do with buildMetadataProperties since we don't have the right MetadataProperties class to plug in. Doing it by param count is a bit sketchy, but works for now, is it worth doing deeper checks about the types or names of the params?

Reply:
hang on, maybe we can plug in the type we got out of buildMetadataProperties

Reply:
Class#getDeclaredMethod

Reply:
cool, got rid of the method loop and changed to declared method